server/src/main/java/org/opensearch/cluster/routing/WeightedRoutingService.java

```diff
@@ -159,9 +160,18 @@ public void verifyAwarenessAttribute(String attributeName) {
         }
     }
 
-    public void ensureNoOngoingDecommissionAction(ClusterState state) {
+    public void ensureNoWeightUpdateForDecommissionedAttribute(ClusterState state, ClusterPutWeightedRoutingRequest request) {
         DecommissionAttributeMetadata decommissionAttributeMetadata = state.metadata().decommissionAttributeMetadata();
-        if (decommissionAttributeMetadata != null && decommissionAttributeMetadata.status().equals(DecommissionStatus.FAILED) == false) {
+        if (decommissionAttributeMetadata != null
+            && decommissionAttributeMetadata.status().equals(DecommissionStatus.FAILED) == false
+            && Objects.equals(
+                request.getWeightedRouting().attributeName(),
+                decommissionAttributeMetadata.decommissionAttribute().attributeName()
+            )
+            && Objects.equals(
+                request.getWeightedRouting().weights().get(decommissionAttributeMetadata.decommissionAttribute().attributeValue()),
+                0.0
+            ) == false) {
             throw new IllegalStateException(
```

Comment:
What http status code would be returned in case of exception?

Reply:
400

Reply:
should we return a [409](https://datatracker.ietf.org/doc/html/rfc7231#section-6.5.8) in this case?

Reply:
Yes. Created a exception throwing 409 due to a state conflict